### PR TITLE
Enable CD

### PR DESCRIPTION
**Pull request by timja**
Also:

- Bump parent pom
- Update bom
- Bump minimum to 2.289.3 (so we are on a supported bom line, integration tests were no longer getting updated as they share a bom version with the plugin)
- Try update deps to new Jenkins version that came with bom upgrade, gave up half way through and disabled enforcer for integration tests

### CODE REVIEW

File: integrations/pom.xml [hunk 1]
```diff
@@ -13,7 +13,8 @@
 
   <properties>
     <maven.deploy.skip>true</maven.deploy.skip> <!-- no need to be deployed during release, this is a test-only module -->
-    <jackson.version>2.12.4</jackson.version>
+    <jackson.version>2.13.1</jackson.version>
+    <enforcer.skip>true</enforcer.skip> <!-- I give up, too much effort to maintain enforcer here, should probably try clear out all enforcer only deps and see if it works -->
```
Comment (jetersen) [inline integrations/pom.xml:17]: 😅 I had actually managed to get rid of it 😭
This is only a PITA so feel free to leave it.

**Review by jetersen — CHANGES_REQUESTED** [review]
.github/workflows/cd.yaml is empty, please fix

File: integrations/pom.xml (hunk 1 above)
Comment (timja) [reply integrations/pom.xml:17]: it just means dependabot PRs fail all the time and such a maintenance PITA

File: .github/dependabot.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ updates:
...
     interval: weekly
   open-pull-requests-limit: 10
-  ignore:
-  - dependency-name: org.jenkins-ci.main:jenkins-core
-    versions:
-    - ">= 0"
+- package-ecosystem: github-actions
+  directory: /
+  schedule:
+    interval: weekly
```
Comment (jetersen) [inline .github/dependabot.yml:11]: Final newline 😓 NIT

```suggestion
    interval: weekly

```

**Review by jetersen — APPROVED** [review]

**Comment by jetersen** [comment]
> dunno why this changed

Plugin loading conflicts?

**Comment by timja** [comment]
> > dunno why this changed
> 
> Plugin loading conflicts?

not that i can see in the log

```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/timja/.m2/repository/org/slf4j/slf4j-simple/1.7.30/slf4j-simple-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/timja/.m2/repository/org/slf4j/slf4j-jdk14/1.7.32/slf4j-jdk14-1.7.32.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
[main] INFO org.eclipse.jetty.util.log - Logging initialized @1062ms to org.eclipse.jetty.util.log.Slf4jLog
=== Starting export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)
   0.066 [id=16]	INFO	o.jvnet.hudson.test.WarExploder#explode: Picking up existing exploded jenkins.war at /Users/timja/code/jenkins/configuration-as-code-plugin/integrations/target/jenkins-for-test
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.Server - jetty-9.4.44.v20210927; built: 2021-09-27T23:02:44.612Z; git: 8da83308eeca865e495e53ef315a249d63ba9332; jvm 11.0.13+8-LTS
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.webapp.StandardDescriptorProcessor - NO JSP Support for /jenkins, did not find org.eclipse.jetty.jsp.JettyJspServlet
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - node0 Scavenging every 660000ms
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.w.WebAppContext@28c1dfcc{Jenkins v2.319.1,/jenkins,file:///Users/timja/code/jenkins/configuration-as-code-plugin/integrations/target/jenkins-for-test/,AVAILABLE}{/Users/timja/code/jenkins/configuration-as-code-plugin/integrations/target/jenkins-for-test}
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.AbstractConnector - Started ServerConnector@48b6f7d{HTTP/1.1, (http/1.1)}{localhost:61853}
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.Server - Started @2326ms
   0.451 [id=16]	INFO	o.jvnet.hudson.test.JenkinsRule#createWebServer: Running on http://localhost:61853/jenkins/
   0.889 [id=30]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
   0.951 [id=39]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$2 (file:/Users/timja/.m2/repository/com/google/inject/guice/4.0/guice-4.0.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$2
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
   5.310 [id=34]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
   5.320 [id=40]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
   5.340 [id=39]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
   5.357 [id=36]	WARNING	jenkins.metrics.api.Metrics#afterExtensionsAugmented: Could not register metrics providers or health check providers as metrics plugin appears to be disabled
   7.028 [id=41]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
   7.293 [id=41]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
   7.293 [id=41]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
   7.306 [id=31]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
   7.458 [id=43]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.Server - jetty-9.4.44.v20210927; built: 2021-09-27T23:02:44.612Z; git: 8da83308eeca865e495e53ef315a249d63ba9332; jvm 11.0.13+8-LTS
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@231e859d{/,null,AVAILABLE}
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.AbstractConnector - Started ServerConnector@c89ebc1{HTTP/1.1, (http/1.1)}{0.0.0.0:61856}
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.Server - Started @9459ms
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.AbstractConnector - Stopped ServerConnector@48b6f7d{HTTP/1.1, (http/1.1)}{localhost:0}
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.session - node0 Stopped scavenging
[Executing export_tool_configuration(io.jenkins.plugins.casc.ToolDefaultPropertiesExportIgnoreListTest)] INFO org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.w.WebAppContext@28c1dfcc{Jenkins v2.319.1,/jenkins,file:///Users/timja/code/jenkins/configuration-as-code-plugin/integrations/target/jenkins-for-test/,STOPPED}{/Users/timja/code/jenkins/configuration-as-code-plugin/integrations/target/jenkins-for-test}
   8.035 [id=16]	INFO	jenkins.model.Jenkins#cleanUp: Stopping Jenkins
   8.294 [id=16]	INFO	jenkins.model.Jenkins#cleanUp: Jenkins stopped

```

**Comment by batmat** [comment]
Awesome work, thanks Tim!

**Pull request merged**